.config/generate_xml.py

```diff
@@ -0,0 +1,65 @@
+import xml.etree.ElementTree as ET
+import yaml
+import sys
+import os
+
+def indent(elem, level=0):
+    '''
+    Nicely formats output xml with newlines and spaces
+    https://stackoverflow.com/a/33956544
+    '''
+    i = "\n" + level*"  "
+    if len(elem):
+        if not elem.text or not elem.text.strip():
+            elem.text = i + "  "
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+        for elem in elem:
+            indent(elem, level+1)
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+    else:
+        if level and (not elem.tail or not elem.tail.strip()):
+            elem.tail = i
+
+try:
+    py_version = sys.argv[1]
+except IndexError:
+    print('No version specified')
+    sys.exit(1)
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+# Load template file
+with open(f'{dir_path}/template.xml', 'r') as f:
+    tree = ET.parse(f)
+    root = tree.getroot()
+
+# Load version dependencies
+with open(f'{dir_path}/{py_version}.yaml', 'r') as f:
+    deps = yaml.safe_load(f)
+
+# Load version and changelog
+with open('jellyfin-kodi/release.yaml', 'r') as f:
+    data = yaml.safe_load(f)
+
+# Populate xml template
+for dep in deps:
+    ET.SubElement(root.find('requires'), 'import', attrib=dep)
+
+# Update version string
+addon_version = data.get('version')
+root.attrib['version'] = f'{addon_version}-{py_version}'
```

Comment:
I believe the Kodi version divider is `~` rather than `-`, but `kodirepo` might take care of that?

Reply:
Scanning through my installed addons on 19, it looks to be largely `+matrix.1`, a few `x.y.z.1`, and a few `-matrix.1`.  I'm not sure it matters, but we can change it something else if we want to attempt to be consistent with the majority of others

Reply:
https://kodi.wiki/index.php?title=Addon.xml#How_versioning_works

https://github.com/oddstr13/python-kodirepo/blob/e857de09218a94a4f9f8efced469ecf6798f765b/kodirepo/__main__.py#L110-L122

https://github.com/xbmc/xbmc/blob/1b4b54e731bf35dd72bd03543e0858feb4f0b85a/xbmc/addons/AddonVersion.cpp#L20L24

`a-zA-Z0-9.+_@~` are the allowed characters for version numbers, so if you've seen `*-matrix`, then that probably wasn't working.

I didn't dig deep enough to figure out whether `kodirepo` re-writes the version in the `addon.xml` file, but I don't think it does, and as such we should change the version string produced here.

I think this would make sense, reserving `~` (`-`) for pre-releases if we ever decide to go that route in the furture (and semver specifies `+` as build metadata, which I think fits here?)
```suggestion
root.attrib['version'] = f'{addon_version}+{py_version}'
```